content/en/blog/trusted-boot.md

```diff
@@ -0,0 +1,155 @@
+
+Throughout these stages, the components—firmware, bootloader, kernel, initrd, and the root file system—work in a sequence to load the operating system successfully. 
+
+This chain is the one we in the Linux and tech industry we are very much familiar with. Several designs and strategies have been developed to secure this stack over time, for instance Verified kernels and drivers, or Secure Boot to sign the bootloader and be able to trust its authenticity. There are various bootloader implementations as well, tied to more secure HW (like TPM devices).
+
+However, as technologies and security measures evolved, attacks evolved as well. Specific hardware now can be leveraged to increase the security posture of the boot process, and it is time for a change, as the market demands more sophisticated security measures to protect against malicious actors.
+
+## Booting with a TPM-equipped Hardware
+
+When a device equipped with TPM, and a sufficiently modern hardware boots, several steps happen sequentially.
+
+Many hardware platforms use a Core Root of Trust for Measurements (CRTM), which is the very first thing that boots, even before the firmware. The CRTM gets a hash of the firmware and sends it to the TPM chip, which measures the running software and is a requirement for trusted boot. The TPM chip then loads the firmware/BIOS. 
+
+The TPM takes note of the measurement and stores the hash in a bank of multiple platform configuration registers (PCRs). In order to store measurements, the TPM chip extends the banks from the previous values, as these changes to the stack are easy to recognize during the boot.
+
+![CRTM trusted boot](https://github.com/kairos-io/kairos-docs/assets/2420543/5a0244c3-ff58-4ea0-83b3-b391ccac0b91)
+
+Next, the BIOS or firmware measures the subsequent stage (bootloader or UKI) and sends it to the TPM, then loads the UKI and continues booting.
+
+Once the TPM is asked to release an encryption key to unseal the full disk encryption, it will check if the measurements it has are valid. If a bootloader is present, it will have already measured the UKI files, and measurement also happens when initrd starts and when the running system is ready to check eventual manipulations (e.g., kernel boot command line). The process is also bound to the secure boot signatures, so any manipulation of the UKI files would be allowed only by the key holders.
+
+If a malicious attacker tries to load custom software or modify an image in any way, they must extend the PCR value and align it with the value it would have had following an expected boot. The cryptographic robustness of the hash algorithm makes achieving this computationally expensive, thereby helping to support security. 
```

Comment:
If an attacker had a way to extend the PCR values, they could make the system trust any modified image, no? That's not computationally expensive.
It would be computationally expensive (or impossible) to modify an image so that it still matches the measurement (if at all possible). Is this correct?

Reply:
you can always extend the PCR values, however, you would have to expand the PCR values in a way that matches the same exact measurements (that's why it's computationally expensive). It's not only a matter of the image, but it's in a more generic sense